Mathlib/Algebra/Module/Submodule/Pointwise.lean

```diff
@@ -277,4 +289,290 @@ scoped[Pointwise] attribute [instance] Submodule.pointwiseMulActionWithZero
+    exact ⟨Finsupp.single r ⟨n, hn⟩, Finsupp.single_mem_supported _ _ hr, by simp⟩
+
+  · intro x hx
+    obtain ⟨c, hc, rfl⟩ := hx
+    simp only [LinearMap.coe_comp, coeSubtype, Finsupp.coe_lsum, Finsupp.sum, LinearMap.coe_mk,
+      AddHom.coe_mk, Function.comp_apply, AddSubmonoid.coe_finset_sum, coe_toAddSubmonoid,
+      SetLike.val_smul]
+    refine Submodule.sum_mem (p := sR • N) (t := c.support) ?_ _ ⟨sR • N, ?_⟩
+    · rintro r hr
+      rw [mem_set_smul_def, Submodule.mem_sInf]
+      rintro p hp
+      exact hp (hc hr) (c r).2
+    · ext x : 1
+      simp only [Set.mem_iInter, SetLike.mem_coe]
+      fconstructor
+      · refine fun h ↦ h fun r n hr hn ↦ ?_
+        rw [mem_set_smul_def, mem_sInf]
+        exact fun p hp ↦ hp hr hn
+      · aesop
+
+lemma mem_set_smul(x : M) [SMulCommClass R R N] :
+    x ∈ sR • N ↔ ∃ (c : R →₀ N), (c.support : Set R) ⊆ sR ∧ x = c.sum fun r m ↦ r • m := by
+  fconstructor
+  · intros h
+    rw [set_smul_eq_map] at h
+    obtain ⟨c, hc, rfl⟩ := h
+    exact ⟨c, hc, rfl⟩
+
+  · rw [mem_set_smul_def, Submodule.mem_sInf]
+    rintro ⟨c, hc1, rfl⟩ p hp
+    simp only [Finsupp.sum, AddSubmonoid.coe_finset_sum, coe_toAddSubmonoid, SetLike.val_smul]
+    exact Submodule.sum_mem _ fun r hr ↦ hp (hc1 hr) (c _).2
+
+@[simp] lemma empty_set_smul : (∅ : Set S) • N = ⊥ := by
+  ext
+  fconstructor
+  · intro hx
+    rw [mem_set_smul_def, Submodule.mem_sInf] at hx
+    exact hx ⊥ (fun r _ hr ↦ hr.elim)
+  · rintro rfl; exact Submodule.zero_mem _
+
+@[simp] lemma set_smul_bot : s • (⊥ : Submodule R M) = ⊥ :=
+  eq_bot_iff.mpr fun x hx ↦ by induction x, hx using set_smul_inductionOn <;> aesop
+
+-- TODO: `r • N` should be generalized to allow `r` to be an element of `S`.
+lemma singleton_set_smul [SMulCommClass R R M] (r : R) :
+    ({r} : Set R) • N = r • N := by
+  apply set_smul_eq_of_le
+  · rintro r m rfl hm; exact ⟨m, hm, rfl⟩
+  · rintro _ ⟨m, hm, rfl⟩
+    rw [mem_set_smul_def, Submodule.mem_sInf]
+    intro p hp; exact hp rfl hm
+
+lemma mem_singleton_set_smul [SMulCommClass R S M] (r : S) (x : M) :
+    x ∈ ({r} : Set S) • N ↔ ∃ (m : M), m ∈ N ∧ x = r • m := by
+  fconstructor
+  · intro hx
+    induction' x, hx using Submodule.set_smul_inductionOn with
+      t n memₜ memₙ t n mem h m₁ m₂ mem₁ mem₂ h₁ h₂
+    · aesop
+    · rcases h with ⟨n, hn, rfl⟩
+      exact ⟨t • n, by aesop,  smul_comm _ _ _⟩
+    · rcases h₁ with ⟨m₁, h₁, rfl⟩
+      rcases h₂ with ⟨m₂, h₂, rfl⟩
+      exact ⟨m₁ + m₂, Submodule.add_mem _ h₁ h₂, by aesop⟩
+    · exact ⟨0, Submodule.zero_mem _, by aesop⟩
+  · rintro ⟨m, hm, rfl⟩
+    aesop
+
+-- Note that this can't be generalized to `Set S`, because even though `SMulCommClass R R M` implies
+-- `SMulComm R R N` for all `R`-submodules `N`, `SMulCommClass R S N` for all `R`-submodules `N`
+-- does not make sense. If we just focus on `R`-submodules that are also `S`-submodule, then this
+-- should be true.
+/-- subset of a ring has a multiplicative action on its submodules-/
```

Comment:
```suggestion
/-- Subset of a ring has a multiplicative action on its submodules. -/
```
This sentence sounds weird to me. In particular "its submodules" seems to be referring to "submodules of R".
I don't have a better alternative though so I'm fine if this is not changed.

Reply:
How about
```suggestion
/-- A subset of a ring `R` has a multiplicative action on submodule of a module over `R`. -/
```